application/Module.php

```diff
@@ -760,6 +775,99 @@ public function noindexItemSet(ZendEvent $event)
         $this->noindexResourceShow($view, $view->itemSet);
     }
 
+    /**
+     * Serialize the API output to JSON-LD.
+     */
+    public function serializeApiOutputJsonLd(ZendEvent $event)
+    {
+        $renderer = $event->getTarget();
+        $model = $event->getParam('model');
+        $format = $event->getParam('format');
+        $payload = $event->getParam('payload');
+        $output = $event->getParam('output');
+
+        if ('jsonld' !== $format) {
+            return;
+        }
+
+        $eventManager = $this->getServiceLocator()->get('EventManager');
+
+        if ($payload instanceof RepresentationInterface) {
+            $args = $eventManager->prepareArgs(['jsonLd' => $output]);
+            $eventManager->trigger('rep.resource.json_output', $payload, $args);
+            $output = $args['jsonLd'];
+        }
+
+        if (null !== $model->getOption('pretty_print')) {
+            // Pretty print the JSON.
+            $output = Json::prettyPrint($output);
+        }
+
+        $jsonpCallback = (string) $model->getOption('callback');
+        if (!empty($jsonpCallback)) {
+            // Wrap the JSON in a JSONP callback. Normally this would be done
+            // via `$this->setJsonpCallback()` but we don't want to pass the
+            // wrapped string to `rep.resource.json_output` handlers.
+            $output = sprintf('%s(%s);', $jsonpCallback, $output);
+            $renderer->setHasJsonpCallback(true);
```

Comment:
We set set `ApiJsonRenderer::$hasJsonpCallback` to true here and I don't know why. Any ideas?

Reply:
It's used to have the content type get set to text/javascript, because JSONP is javascript code rather than JSON.